### PR TITLE
Feat: Round Validation System Implementation

### LINKED ISSUE #126: Feat: Round Validation System Implementation
## Overview
Implement the comprehensive validation system for Round configuration. This issue covers the static validation method that ensures round configurations are valid before creation.

#### `validate_config()` - Static Configuration Validation
```rust
fn validate_config(config: @RoundConfig) -> RoundValidation
```
- Static method in `RoundTrait` for validating `RoundConfig`
- Comprehensive validation of all configuration parameters
- Return `RoundValidation` with success/failure and error messages
- Used during round creation to ensure valid configurations

### Player Count Validation
- `max_players` must be > 0 (cannot be zero)
- `max_players` must be ≤ 50 (system limit)
- Solo mode should typically have max_players = 1

### Cards Configuration Validation
- `cards_per_round` must be > 0 (cannot be zero)
- `cards_per_round` must be ≤ 100 (system limit)
- Reasonable card counts for game balance

### Timeout Validation
- `card_timeout` must be > 0 (cannot be zero)
- Reasonable timeout values (suggest minimum practical values)

### Challenge Type Parameter Validation
- If `challenge_type` is provided, validate required parameters
- `requires_param()`: Types that need `challenge_param1`
- `requires_two_params()`: Types that need both parameters
- Parameter validation based on challenge type requirements

### Mode-Specific Validation
- Validate mode-specific constraints
- Wager mode validation (if applicable)
- Solo mode constraints

## Acceptance Criteria
- [ ] `validate_config()` method implemented correctly
- [ ] All validation rules properly enforced
- [ ] Comprehensive parameter validation for challenge types
- [ ] Descriptive error messages for all failure cases
- [ ] Success case returns proper validation result
- [ ] Complete test coverage for all validation scenarios
- [ ] All edge cases and boundary values tested

## Note

**Target Branch**
- `refac/modular-architecture`

**File Location**
- `src/models/round.cairo`

**Target Trait**
- `RoundTrait`

### PR DESCRIPTION
**Pull request by No-bodyq**
closes #126

This PR adds comprehensive validation logic for the RoundConfig struct through the new validate_config function, along with a full suite of unit tests to ensure proper configuration rules are enforced.

## Changes Made
- **Implemented `validate_config`** to check for invalid game configuration values:
  - Wager amount cannot be zero.
  - Solo mode must have exactly 1 player.
  - Player count must be between 1 and 50.
  - Cards per round must be between 1 and 100.
  - Card timeout must be greater than zero.
  - Challenge type/parameter validation:
    - Enforces required params based on challenge type.
    - Rejects extra params for challenges that don’t require them.
    - Ensures param values are non-zero when required.
    - Prevents params from being set for `Random` challenges.
    - Enforces both params for `GenreAndDecade` challenges.
-  Created `create_base_config` helper to simplify test setup.
- Added full test coverage for all validation rules.

## Testing
- Added unit tests covering all possible validation cases.
- Verified valid configurations pass and all invalid configurations return the correct error message.

### CODE REVIEW

**Comment by No-bodyq** [comment]
Kindly review @manlikeHB

File: src/models/round/round.cairo [hunk 1]
```diff
@@ -78,6 +78,118 @@ pub impl RoundImpl of RoundTrait {
     fn get_challenge_type(self: @Round) -> ChallengeType {
         (*self.challenge_type).try_into().unwrap()
     }
+
+    fn validate_config(config: @RoundConfig) -> RoundValidation {
+        let mut is_valid = true;
+        let mut error_message: felt252 = 0;
+
+        if *config.wager_amount == 0 {
```
Comment (manlikeHB) [inline src/models/round/round.cairo]: This should only apply to the `Mode::WagerMultiPlayer`

**Review by manlikeHB — CHANGES_REQUESTED** [review]

**Review by manlikeHB — APPROVED** [review]
Thank you! LGTM!
